src/amo/components/GuidesAddonCard/index.js

```diff
@@ -115,29 +86,23 @@ export class GuidesAddonCardBase extends React.Component<InternalProps> {
                 {this.props.addonCustomText}
               </p>
             </div>
-            {showInstallButton && currentVersion && (
-              <AMInstallButton
+            {addon && (
+              <InstallButtonWrapper
                 addon={addon}
-                currentVersion={currentVersion}
+                className="guides"
                 defaultButtonText={i18n.gettext('Add')}
-                defaultInstallSource={this.props.defaultInstallSource}
-                disabled={!compatible}
-                enable={this.props.enable}
-                hasAddonManager={this.props.hasAddonManager}
-                install={this.props.install}
-                installTheme={this.props.installTheme}
-                setCurrentStatus={this.props.setCurrentStatus}
-                status={this.props.installStatus}
-                uninstall={this.props.uninstall}
-                isAddonEnabled={this.props.isAddonEnabled}
+                defaultInstallSource={defaultInstallSource}
+                enable={enable}
+                getFirefoxButtonType={GET_FIREFOX_BUTTON_TYPE_ADDON}
+                hasAddonManager={hasAddonManager}
+                install={install}
+                installTheme={installTheme}
+                isAddonEnabled={isAddonEnabled}
                 puffy={false}
+                setCurrentStatus={setCurrentStatus}
+                uninstall={uninstall}
```

Comment:
I am guessing not if it wasn't done this way initially but I guess I am still curious..

Reply:
I originally wanted to do that, but I thought there was a problem because we still need to wrap `Addon` with the `withInstallHelpers` HOC for disco pane. I realize now that we don't use this new `InstallButtonWrapper` for disco pane, so I guess we can do that. Let me give it a try and see if any other issues come up.

Reply:
I managed to get all of it working, except for the `defaultInstallSource` part, which is weird because it seems that we need to pass a value for that when we're applying the HOC, which won't work in this case because it can have different values in `InstallButtonWrapper`. I'm not sure if there's a simple solution to this, or if I need to make some changes to the HOC to deal with this. I'm going to push a new commit anyway so you can take a look at what I have thus far.

Reply:
Ok, I got it all working!

Reply:
*maybe* we could do something similar to how the `withFixedErrorHandler` HOC passes in props..? For example I see this in some examples (I modified slightly for this case):

```javascript
export const getInstallSource = (ownProps: InternalProps) => {
  return ownProps.defaultInstallSource;
};

const InstallButtonWrapper: React.ComponentType<Props> = compose(
  ...
  withInstallHelpers({ defaultInstallSource: getInstallSource }), // can we pass it in this way or get it this way somehow?
)(InstallButtonWrapperBase);
```

Does that make any sense? this might be more work tho so maybe could be done in a follow up issue if you see it has value.